js/game.js

```diff
@@ -0,0 +1,253 @@
+  }
+
+  createPlayers() {
+    return [
+      new Player('Player 1', 'cross', true),
+      new Player('Player 2', 'circle')
+    ];
+
+  }
+
+  get activePlayer() {
+    return this.players.find(player => player.active);
+  }
+
+  get unactivePlayer() {
+    return this.players.find(player => !player.active);
+  }
+
+  switchPlayers() {
+    this.players.forEach(player => player.active = !player.active);
+  }
+
+  startGame(boardDiv) {
+    this.board.renderFields(boardDiv);
+    this.ready = true;
+  }
+
+  makeMove(x,y){
+    const element = document.querySelector(`[data-x="${x}"][data-y="${y}"]`);
+    const field = this.board.fields[x][y];
+    field.free = false;
+    field.owner = this.activePlayer;
+    field.renderSymbol(element);
+    this.fieldsLeft--;
+
+    if (this.checkForWin(this.activePlayer)){
+      alert(`${this.activePlayer.name} wins`);
+    }else if (this.fieldsLeft === 0) {
+      alert('draw!');
+    }else{
+      this.switchPlayers();
+      if (this.activePlayer.name === 'Player 2') this.computerMove();
+      this.ready = true;
+    }
+  }
+
+  handleInteraction(e) {
+    if (this.ready){
+      this.ready = false;
+      const element = e.target;
+      const x = element.dataset.x;
+      const y = element.dataset.y;
+      const field = this.board.fields[x][y];
+      if (!field.free) return;
+      else {
+        this.makeMove(x,y);
+      }
+    }
+  }
+
+  checkForWin(player) {
+    if(
+        (
+          this.board.fields[0][0].owner === player&&
+          this.board.fields[0][1].owner === player&&
+          this.board.fields[0][2].owner === player
+        )
+        ||
+        (
+          this.board.fields[1][0].owner === player&&
+          this.board.fields[1][1].owner === player&&
+          this.board.fields[1][2].owner === player
+        )
+        ||
+        (
+          this.board.fields[2][0].owner === player&&
+          this.board.fields[2][1].owner === player&&
+          this.board.fields[2][2].owner === player
+        )
+        ||
+        (
+          this.board.fields[0][0].owner === player&&
+          this.board.fields[1][0].owner === player&&
+          this.board.fields[2][0].owner === player
+        )
+        ||
+        (
+          this.board.fields[0][1].owner === player&&
+          this.board.fields[1][1].owner === player&&
+          this.board.fields[2][1].owner === player
+        )
+        ||
+        (
+          this.board.fields[0][2].owner === player&&
+          this.board.fields[1][2].owner === player&&
+          this.board.fields[2][2].owner === player
+        )
+        ||
+        (
+          this.board.fields[0][0].owner === player&&
+          this.board.fields[1][1].owner === player&&
+          this.board.fields[2][2].owner === player
+        )
+        ||
+        (
+          this.board.fields[0][2].owner === player&&
+          this.board.fields[1][1].owner === player&&
+          this.board.fields[2][0].owner === player

```

Comment:
Ten fragment jest kompletnie nie czytelny przez liczbę `or`, która tutaj stosujesz i generalnie ilość warunków. Trzeba to uprościć i przerobić. Pomyślę jeszcze nad tym

Reply:
powiem Ci szczerze, to jest bardzo podobne, do tego co bylo na treehouse, co sie zastanawialem jak napisac warunki okreslajce win i tam uzyli takiego samego piekiełka. w wypadku kólka i krzyzyk, jest o tyle latwiej ze masz tylko 8 ukladow boarda które wygrywaja. nie wiem, moze jak kod ma być uzywany potem to w tym wypadku komentarz by wystarczył ? to jest rzecz, ktora wg mnie nie moze sie zmienic bo jest esencja gry

Reply:
Pamiętam, że na Treehouse robili coś podobnego. W takim wypadku komentarz zdecydowanie jest dobrym rozwiązaniem, z krótkim wyjaśnieniem jak działa ten warunek / warunki. Pomyślę jeszcze o tym, bo wydaje mi się, że da się to zautomatyzować ;)